modules/programs/wezterm.nix

```diff
@@ -0,0 +1,589 @@
+      options = {
+        foreground = mkColor "#C0C0C0" "The default text color.";
+        background = mkColor "#000000" "The default background color.";
+
+        scrollbarThumb = mkColor "#222222" ''
+          The color of the scrollbar "thumb"; the portion that represents the current viewport.'';
+        split = mkColor "#444444" "The color of the split lines between panes";
+
+        cursor = mkOption {
+          type = weztermColorschemeCursorType;
+          default = { };
+          description = ''
+            The color of the cursor.
+          '';
+          example = literalExample ''
+            {
+              foreground = "#52ad70";
+              background = "#000000";
+              border = "#52AD70";
+            }
+          '';
+        };
+
+        selection = mkOption {
+          type = weztermColorschemeSelectionType;
+          default = { };
+          description = ''
+            The color of the selection.
+          '';
+          example = literalExample ''
+            {
+              foreground = "#FFFACD";
+              background = "#000000";
+            }
+          '';
+        };
+
+        tabBar = mkOption {
+          type = weztermColorschemeTabBarType;
+          default = { };
+          description = ''
+            The colors of the tab bar.
+          '';
+          example = literalExample ''
+            {
+              background = "#0b0022";
+            }
+          '';
+        };
+
+        ansi = mkOption {
+          type = weztermColorschemeBase16Type true;
+          default = { };
+          description = ''
+            The main ANSI colors.
+          '';
+          example = literalExample ''
+            {
+              black = "#000000";
+              red = "#800000";
+              green = "#008000";
+              yellow = "#808000";
+              blue = "#000080";
+              magenta = "#800080";
+              cyan = "#008080";
+              white = "#C0C0C0";
+            }
+          '';
+        };
+
+        bright = mkOption {
+          type = weztermColorschemeBase16Type false;
+          default = { };
+          description = ''
+            The main bright colors.
+          '';
+          example = literalExample ''
+            {
+              black = "#808080";
+              red = "#FF0000";
+              green = "#00FF00";
+              yellow = "#FFFF00";
+              blue = "#0000FF";
+              magenta = "#FF00FF";
+              cyan = "#00FFFF";
+              white = "#FFFFFF";
+            }
+          '';
+        };
+      };
+    };
+
+in {
+  meta.maintainers = with lib.hm.maintainers; [ l3af ];
+
+  options.programs.wezterm = with types; {
+    enable = mkEnableOption "Wezterm terminal emulator";
+
+    package = mkOption {
+      type = package;
+      default = pkgs.wezterm;
+      defaultText = "pkgs.wezterm";
+      description = ''
+        Wezterm package to use. Set to <code>null</code> to use default package.
```

Comment:
Does setting the package to `null` actually work?

Reply:
It doesn't anymore, I originally had a check to see if it was set to null but that has been removed so that should be updated.